Objects/codeobject.c

```diff
@@ -1894,21 +1894,9 @@ _PyHydra_UnicodeFromOffset(struct lazy_pyc *pyc, uint32_t offset)
 PyObject *
 _PyHydra_UnicodeFromIndex(struct lazy_pyc *pyc, int index)
 {
-    //fprintf(stderr, "   >>>>    _PyHydra_UnicodeFromIndex index = %d pyc->n_strings=%d\n", index, pyc->n_strings);
     if (0 <= index && index < pyc->n_strings) {
-        uint32_t offset = pyc->string_offsets[index];
-        //fprintf(stderr, "[1]         offset = %d\n", offset);
-        if (offset & 1) {
-            index = offset >> 1;
-            assert(0 <= index && index < pyc->n_strings);
-            offset = pyc->string_offsets[index];
-            assert(!(offset & 1));
-            // fprintf(stderr, "[2]         offset = %d\n", offset);
-        }
+        uint32_t offset = LAZY_CO_NAME_OFFSET(pyc, index);
         PyObject *res = _PyHydra_UnicodeFromOffset(pyc, offset);
```

Comment:
Question— why not have the redirect handling in the FromOffset functions, too?

Reply:
It's not always needed - blob_offsets and code_offsets can't have any redirects.